storage/remote/codec_test.go

```diff
@@ -116,24 +106,16 @@ func TestValidateLabelsAndMetricName(t *testing.T) {
 				"label2", "name",
 			),
 			expectedErr: "invalid label name: ",
-			shouldPass:  false,
 			description: "don't report as duplicate label name",
 		},
 	}
 
 	for _, test := range tests {
 		t.Run(test.description, func(t *testing.T) {
 			err := validateLabelsAndMetricName(test.input)
-			if err == nil {
-				if !test.shouldPass {
-					t.Fatalf("Test should fail, but passed instead.")
-				}
-			} else {
-				if test.shouldPass {
-					t.Fatalf("Test should pass, got unexpected error: %v", err)
-				} else if err.Error() != test.expectedErr {
-					t.Fatalf("Test should fail with: %s got unexpected error instead: %v", test.expectedErr, err)
-				}
+			if test.expectedErr != "" || err != nil {
```

Comment:
Though this works, it's not easy to grasp at first glance. Can you separate it out like this `if test.expectedErr != "" {...} else {...}`? (failure message also gets better like this)

Reply:
done! WDYT?